scripts/ai_generation/converter.py

```diff
@@ -0,0 +1,461 @@
+            if end_match:
+                end_index = before_you_begin_match.end() + end_match.start()
+            else:
+                end_index = len(text)
+        else:
+            end_match = re.search(r"^#+\s.*$", text[section_match.end():], re.MULTILINE)
+            if end_match:
+                end_index = section_match.end() + end_match.start()
+            else:
+                end_index = len(text)
+        if len(env_var_dict) > 0:
+            env_vars = ""
+            for key in env_var_dict:
+                env_vars += f"\nexport {key}={env_var_dict[key]}"
+        
+            text = text[:end_index] + "\n" + f"""## Define Environment Variables\n\nThe First step in this tutorial is to define environment variables.\n\n```bash\nexport RANDOM_ID="$(openssl rand -hex 3)"{env_vars}\n```\n"""+text[end_index:]
+    
+    return text
+
+def get_environment_vars(text):
+    code_dict = {}
+    env_var_dict = {}
+    commands = re.findall(r'```[\w\W]*?```', text)
+    for raw_command in commands:
+        command_type = [element for element in allowed_commands_list if element in raw_command]
+        if len(command_type) > 0: 
+          pattern = r"--(\S+)\s+(\S+)"
+          matches = re.findall(pattern, raw_command)
+          if len(matches) > 0:
+            for match in matches:
+                key, value = match[0], match[1]
+                if key in code_dict and value not in code_dict[key]:
+                    code_dict[key].append(value)
+                elif key in code_dict and value in code_dict[key]:
+                    continue
+                else:
+                    code_dict[key] = [value]
+    for key in code_dict:
+        for index, value in enumerate(code_dict[key]):
+            if len(code_dict[key]) > 1:
+              env_key = ''.join([word.capitalize() for word in key.split('-')])  
+              env_var_dict[f'My{env_key}{index+1}'] = value
+            else:
+              env_key = ''.join([word.capitalize() for word in key.split('-')])  
+              env_var_dict[f'My{env_key}'] = value
+    return env_var_dict
+
+def get_azure_doc_text(url): 
+    response = requests.get(url) 
+    return(response.text) 
+
+def get_latest_error_log():
+    error_line_num = 0
+    log_file = 'ie.log'
+    with open(log_file, 'r') as file:
+        for i, line in enumerate(file, 1):
+            if re.search(r'level=error', line):
+                error_line_num = i
+
+    lines_from_error = []
+    with open(log_file, 'r') as file:
+        for i, line in enumerate(file, 1):
+            if i >= error_line_num:
+                lines_from_error.append(line)
+
+    return ' '.join(lines_from_error)
+
+def create_pr(dirname):  
+    os.environ['GITHUB_TOKEN'] = github_access_token
+    branch_name = f"aiexecdocs/{dirname}" # os.popen('git rev-parse --abbrev-ref HEAD').read().strip()
+    label = "ai-doc"
+    subprocess.run(f'git checkout -b {branch_name}', shell=True)
```

Comment:
We should add error handling to make sure that if one of these subprocesses fails early, we can let the user know when/which one.